sram.c

```diff
@@ -0,0 +1,96 @@
+#include "sram.h"
+#include "spi.h"
+#include "nuts_bolts.h"
+
+/* Least and most significant bytes of x, of type uint16_t */
+#define LSB(x) (x & 0x00FF)
+#define MSB(x) ((x & 0xFF00) >> 8)
+
+#define MODE_MASK 0xC0
+#define MODE_IDX  6U
+
+/* SPI insructions */
+enum instruction_e {
+  READ = 0x3,
+  WRITE = 0x2,
+  EDIO = 0x3B,
+  EQIO = 0x38,
+  RSTIO = 0xFF,
+  RDMR = 0x5,
+  WRMR = 0x1
+};
+
+void sram_init()
+{
+  /* Configure SCS pin as output */
+  SCS_SRAM_DDR |= (1 << SCS_SRAM_DDR_PIN);
+
+  /* Set SCS pin to high. The CS for this chip is active low */
+  SCS_SRAM_PORT &= ~(1 << SCS_SRAM_PIN);
+
+  /* SCK resting state is 0. Clock data on rising edge */
+  spi_set_mode(0, 0);
+
+  sram_set_mode(BYTE_MODE);
+}
+
+uint8_t sram_read_mode()
+{
+  spi_set_mode(0, 0);
+
+  uint8_t data_out[2] = {RDMR, 0xFF};
```

Comment:
The functions `sram_read_mode` and `sram_set_mode` are identical except for the right-hand side of this line and the return value (one of them returns nothing). Is there a way they can be unified into a single function? Would it make semantic sense to do that?